src/sentry/tasks/integrations/github/link_all_repos.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import sentry_sdk
+
+from sentry.models.organization import Organization
+from sentry.plugins.base import bindings
+from sentry.services.hybrid_cloud.integration import integration_service
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.tasks.base import instrumented_task
+from sentry.tasks.integrations.github.pr_comment import RATE_LIMITED_MESSAGE
+from sentry.utils import metrics
+
+logger = logging.getLogger(__name__)
+
+
+@instrumented_task(name="sentry.integrations.github.link_all_repos")
+def link_all_repos(integration_id: int, organization_id: int):
+    integration = integration_service.get_integration(integration_id=integration_id)
+    if not integration:
+        logger.error(
+            "github.link_all_repos.integration_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr("github.link_all_repos.error", tags={"type": "missing_integration"})
+        return
+
+    try:
+        organization = Organization.objects.get(id=organization_id)
+    except Organization.DoesNotExist:
+        logger.error(
+            "github.link_all_repos.orgnanization_missing",
+            extra={"organization_id": organization_id},
+        )
+        metrics.incr("github.link_all_repos.error", tags={"type": "missing_organization"})
+        return
+
+    installation = integration_service.get_installation(
+        integration=integration, organization_id=organization_id
+    )
+
+    gh_client = installation.get_client()
+
+    try:
+        gh_repositories = gh_client.get_repositories(fetch_max_pages=True)
+    except ApiError as e:
+        if e.json and RATE_LIMITED_MESSAGE in e.json.get("message", ""):
+            metrics.incr("github.link_all_repos.rate_limited_error")
+            return
+
+        metrics.incr("github.link_all_repos.api_error")
+        raise e
+
+    binding_key = "integration-repository.provider"
+    provider_cls = bindings.get(binding_key).get("integrations:" + integration.provider)
+    provider = provider_cls(id=integration.provider)
```

Comment:
Is this method of getting the the provider consistent across different repository integrations? (e.g. Github, Gitlab, Bitbucket) If so, it might be helpful to pull this into a method on the `RepositoryMixin` or `IntegrationInstallation`

Reply:
Also, even if you need to `type: ignore` it, I'd add a type to `provider`  and `provider_cls` since these mixin classes can be many different shapes at run time.

Reply:
this is done across a couple of places in the codebase. i referenced https://github.com/getsentry/sentry/blob/master/src/sentry/tasks/commits.py#L109-L115 for this

this appears to be consistent across all repos https://github.com/getsentry/sentry/blob/master/src/sentry/models/repository.py#L52-L60